examples/inverse/plot_use_sourcemorph.py

```diff
@@ -0,0 +1,178 @@
+
+raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
+event_fname = data_path + '/MEG/sample/sample_audvis_raw-eve.fif'
+
+t1_sample = data_path + '/subjects/sample/mri/brain.mgz'
+t1_fsaverage = data_path + '/subjects/fsaverage/mri/brain.mgz'
+
+###############################################################################
+# Compute example data. For reference see :ref:`lcmv beamformer
+# <sphx_glr_auto_examples_inverse_plot_lcmv_beamformer.py>` and
+# :ref:`lcmv beamformer volume
+# <sphx_glr_auto_examples_inverse_plot_lcmv_beamformer_volume.py>`
+
+fname_fwd = data_path + '/MEG/sample/sample_audvis-meg-eeg-oct-6-fwd.fif'
+label_name = 'Aud-lh'
+fname_label = data_path + '/MEG/sample/labels/%s.label' % label_name
+event_id, tmin, tmax = [1, 2], -0.2, 0.2
+
+# Setup for reading the raw data
+raw = mne.io.read_raw_fif(raw_fname, preload=True)
+raw.info['bads'] = ['MEG 2443', 'EEG 053']  # 2 bads channels
+events = mne.read_events(event_fname)
+
+# Set up pick list: EEG + MEG - bad channels (modify to your needs)
+picks = mne.pick_types(raw.info, meg=True, eeg=False, stim=True, eog=True,
+                       exclude='bads')
+
+# Pick the channels of interest
+raw.pick_channels([raw.ch_names[pick] for pick in picks])
+
+# Re-normalize our empty-room projectors, so they are fine after subselection
+raw.info.normalize_proj()
+
+# Read epochs
+epochs = mne.Epochs(raw, events, event_id, tmin, tmax,
+                    baseline=(None, 0), preload=True, proj=True,
+                    reject=dict(grad=4000e-13, mag=4e-12, eog=150e-6))
+evoked = epochs.average()
+
+# Read forward model
+forward = mne.read_forward_solution(fname_fwd)
+forward_surf = mne.convert_forward_solution(forward, surf_ori=True)
+
+# Compute regularized noise and data covariances
+noise_cov = mne.compute_covariance(epochs, tmin=tmin, tmax=0, method='shrunk')
+data_cov = mne.compute_covariance(epochs, tmin=0.04, tmax=0.15,
+                                  method='shrunk')
+
+# Compute unit-noise-gain beamformer with whitening of the leadfield and
+# data (enabled by passing a noise covariance matrix)
+filters = make_lcmv(evoked.info, forward_surf, data_cov, reg=0.05,
+                    noise_cov=noise_cov, pick_ori='max-power',
+                    weight_norm='nai')
+
+# Apply this spatial filter to source-reconstruct the evoked data
+stc_surf = apply_lcmv(evoked, filters, max_ori_out='signed')
+
+# To save memory
+stc_surf.crop(0.087, 0.087)
+
+# Compute VolSourceEstimate
+fname_fwd = data_path + '/MEG/sample/sample_audvis-meg-vol-7-fwd.fif'
+
+# Read forward model
+forward_vol = mne.read_forward_solution(fname_fwd)
+
+filters = make_lcmv(evoked.info, forward_vol, data_cov, reg=0.05,
+                    noise_cov=noise_cov, pick_ori='max-power',
+                    weight_norm='nai')
+
+# Apply this spatial filter to the evoked data.
+stc_vol = apply_lcmv(evoked, filters, max_ori_out='signed')
+
+# To save memory
+stc_vol.crop(0.087, 0.087)
```

Comment:
I feel that the purpose here is to illustrate the SourceMorph class. Therefore you just want a volume STC or surface STC. I would just use dSPM with precomputed stuff from sample dataset has much as possible so you need much less lines. Here the core of the examples starts line 118.

Reply:
+1 same feeling here